File: src/main/java/com/neutralplasma/virtusbot/handlers/playerLeveling/MultiplierData.java
package com.neutralplasma.virtusbot.handlers.playerLeveling;

public class MultiplierData {
    private int multiplier;
    private long endsOn = 0;


    public int getMultiplier() {
        if(isActive()){
            return multiplier;
        }else{
            return 1;
        }
    }

    public void setMultiplier(int multiplier, long time){
        this.multiplier = multiplier;
        this.endsOn = System.currentTimeMillis() + time;
    }

    public boolean isActive(){
        return endsOn - System.currentTimeMillis() > 0;
    }
}

File: src/main/java/com/neutralplasma/virtusbot/handlers/playerLeveling/MultiplierData.kt
package com.neutralplasma.virtusbot.handlers.playerLeveling

data class MultiplierData(
        var multiplier: Int,
        var endsOn: Long,

){

    private val isActive: Boolean
        get() = endsOn - System.currentTimeMillis() > 0

    fun setMultiplier(multiplier: Int, endsOn: Long){
        this.multiplier = multiplier;
        this.endsOn = endsOn + System.currentTimeMillis();
    }
    fun getActiveMultiplier(): Int {
        if (isActive){
            return this.multiplier;
        }
        return 1;
    }
}